pkg/service/cluster/model.go

```diff
@@ -19,6 +19,16 @@ type Cluster struct {
 	KnownHosts []string  `json:"-"`
 	Port       int       `json:"port,omitempty"`
 	AuthToken  string    `json:"auth_token"`
+	// Three values are needed here where tls_enabled = Null means that it's unknown
+	// if the cluster has this flag enabled or not. It's managers responsibility to
+	// find the correct value on the first usage, basing on the already existing logic where scylla configuration
+	// is checked against `/config/client_encryption_options` API call to scylla server.
+	// Null value is introduced to seamlessly support migration period.
+	// Beginning Scylla Manager 3.2.6, it's user responsibility to explicitly set on cluster
+	// if the healtchecks should be performed on encrypted or unencrypted connection.
+	//
+	// https://github.com/scylladb/scylla-manager/issues/3679
+	TLSEnabled *bool `json:"tls_enabled"`
```

Comment:
I don't fully understand the need for 3 values here - why can't we just have a single flag `--disable-tls` (default false)?

Also, I don't think that this approach solves problem with choosing correct port described [here](https://github.com/scylladb/scylla-manager/pull/3699#issuecomment-1912025255) (SM can't tell whether TLS is listened on regular or ssl port).

There are basically 3 things that SM needs to know:
- which port should it use?
- should it use encryption?
- should it provide certs?

It has been shown that there is no way for SM to guess the correct port without dialing it, so this information should be passed by user (e.g. boolean `--disable-ssl-port`, default false, or maybe `--port` which allows for overwriting used port with either a number or a "native" / "native-ssl" strings.)

Even when we know the port (regular or ssl), SM doesn't always know whether this port is encrypted. So it indeed needs a flag for that (e.g. `--disable-encryption`, default false)

In terms of the third question, we are lucky that at least `ClientEncryptionRequireAuth` seems like a reliable source of truth, so SM can just take it from Scylla config.

Reply:
3 values are added to smoothly go from the previous approach where the manager were defining if the TLS should be used or not just during the session creation. 
Other approach would be to just migrate all the clusters to update the infromation about disabled/enabled TLS in the DB on manager upgrade. Wanted to avoid it, go migration code is executed outside of the context of manager. There is no handlers to Scylla client available for example. 
https://github.com/scylladb/scylla-manager/tree/master/pkg/schema/migrate

So, if the TLSEnabled = null, then it will failback to this simple check. So the TLSEnabled will be equal scylla configuration (ClientEncryptionEnabled)
https://github.com/scylladb/scylla-manager/blob/35addef69c0eb2b2012ef4367548e05f4daac5b1/pkg/service/cluster/service.go#L578-L589

> which port should it use?

if tls_enabled -> https://github.com/scylladb/scylla-manager/blob/35addef69c0eb2b2012ef4367548e05f4daac5b1/pkg/service/cluster/service.go#L592
otherwise https://github.com/scylladb/scylla-manager/blob/35addef69c0eb2b2012ef4367548e05f4daac5b1/pkg/service/cluster/service.go#L553

> should it use encryption?

It's exactly the "tls_enabled" property of the cluster (which can be nil)

> should it provide certs?

https://github.com/scylladb/scylla-manager/blob/35addef69c0eb2b2012ef4367548e05f4daac5b1/pkg/service/cluster/service.go#L598

> It has been shown that there is no way for SM to guess the correct port without dialing it, so this information should be passed by user (e.g. boolean --disable-ssl-port, default false, or maybe --port which allows for overwriting used port with either a number or a "native" / "native-ssl" strings.)

This is part of `node_info` call response. The call is made to scylla API through agent proxy. 
We don't need to store it in DB. I don't want to store it in DB, lets the manager auto-configure itself whenever there is a change in scylla config. By autoconfigure, I mean just dynamically read it everytime when it's needed.

> Even when we know the port (regular or ssl), SM doesn't always know whether this port is encrypted. So it indeed needs a flag for that (e.g. --disable-encryption, default false)

It's part of node_info response.

There is no intention to completly remove node_info call to agent here.

Reply:
To sum up:
- tls_enabled=true explicitly set to false mean **Enable TLS no matter what scylla says about its configuration**
- tls_enabled=false explicitly set to false mean **Disable TLS no matter what scylla says about its configuration**
- tls_enabled=NIL explicitly set to false mean **I have no idea, please check scylla-configuration**

Reply:
Removed `--tls-enabled` and replaced with two flags `--force-tls-disabled` and `--force-non-ssl-port`